src/plugins/dashboard/public/dashboard_actions/add_agg_vis_action.ts

```diff
@@ -0,0 +1,55 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import { i18n } from '@kbn/i18n';
+import { EmbeddableApiContext } from '@kbn/presentation-publishing';
+import { COMMON_EMBEDDABLE_GROUPING } from '@kbn/embeddable-plugin/public';
+import { Action } from '@kbn/ui-actions-plugin/public';
+import { DASHBOARD_APP_ID } from '../dashboard_constants';
+import { pluginServices } from '../services/plugin_services';
+
+const ADD_AGG_VIS_ACTION_ID = 'ADD_AGG_VIS';
+
+export class AddAggVisualizationPanelAction implements Action<EmbeddableApiContext> {
+  public readonly type = ADD_AGG_VIS_ACTION_ID;
+  public readonly id = ADD_AGG_VIS_ACTION_ID;
+  public readonly grouping = [COMMON_EMBEDDABLE_GROUPING.legacy];
+
+  private showNewVisModal;
+
+  constructor() {
+    ({
+      visualizations: { showNewVisModal: this.showNewVisModal },
+    } = pluginServices.getServices());
+  }
+
+  public getIconType() {
+    return 'visualizeApp';
+  }
+
+  public getDisplayName() {
+    return i18n.translate('dashboardAction.addAggVis.displayName', {
+      defaultMessage: 'Aggregation based',
+    });
+  }
+
+  public async isCompatible() {
+    return true;
```

Comment:
Need further clarification about this, I'm assuming it would be compatible since this action only calls a function to display the selection modal.

Reply:
I would check that at least `apiCanAddNewPanel(embeddable)` is true to make sure the embeddable is capable of adding new panels.

Reply:
FYI you would be checking to ensure the **parent API** is compatible - so something like `apiHasParentApi(embeddable) && apiCanAddNewPanel(embeddable.parentApi)` :)